File: gherkin/dart/lib/src/ast/Comment.dart
import 'package:gherkin/core.dart';
import 'package:gherkin/extensions.dart';
import 'package:gherkin/language.dart';

class Comment implements INullSafetyObject
{
  static const empty = _InvalidComment();

  final Location location;

  final String text;

  const Comment(this.location, this.text);

  @override
  bool get isEmpty => false;

  @override
  bool get isNotEmpty => !isEmpty;
}

/// Convenience implementation of an invalid [Comment] instance.
class _InvalidComment extends Comment
{
  const _InvalidComment()
      : super(Location.empty, Strings.empty);

  @override
  bool get isEmpty => true;
}

File: gherkin/dotnet/Gherkin/Ast/Comment.cs
namespace Gherkin.Ast
{
    public class Comment : IHasLocation
    {
        public Location Location { get; private set; }
        public string Text { get; private set; }

        public Comment(Location location, string text)
        {
            Text = text;
            Location = location;
        }
    }
}